core/server/policy_violations_test.go

```diff
@@ -0,0 +1,230 @@
+	g.Expect(res.Validation).NotTo(BeNil())
+	g.Expect(res.Validation.Id).To(Equal("55101548-12c1-4f79-a09a-a12979903f"))
+	g.Expect(res.Validation.Name).To(Equal("Test Policy"))
+	g.Expect(res.Validation.PolicyId).To(Equal("weave.policies.test-policy"))
+	g.Expect(res.Validation.ClusterId).To(Equal("cluster-1"))
+	g.Expect(res.Validation.Category).To(Equal("Access Control"))
+	g.Expect(res.Validation.Severity).To(Equal("high"))
+	g.Expect(res.Validation.CreatedAt).To(Equal("0001-01-01T00:00:00Z"))
+	g.Expect(res.Validation.Message).To(Equal("Policy event"))
+	g.Expect(res.Validation.Entity).To(Equal("my-deployment"))
+	g.Expect(res.Validation.Namespace).To(Equal("weave-system"))
+	g.Expect(res.Validation.Description).To(Equal("Missing app label"))
+	g.Expect(res.Validation.HowToSolve).To(Equal("how_to_solve"))
+	g.Expect(res.Validation.ViolatingEntity).To(Equal(`{"apiVersion":"apps/v1","kind":"Deployment","metadata":{"name":"nginx-deployment","namespace":"default","uid":"af912668-957b-46d4-bc7a-51e6994cba56"},"spec":{"template":{"spec":{"containers":[{"image":"nginx:latest","imagePullPolicy":"Always","name":"nginx","ports":[{"containerPort":80,"protocol":"TCP"}]}]}}}}`))
+	g.Expect(res.Validation.ClusterName).To(Equal("Default"))
+	g.Expect(res.Validation.Occurrences).To(Equal([]*pb.PolicyValidationOccurrence{{Message: "occurrence details"}}))
+
+	// non existing validation
+	_, err = c.GetPolicyValidation(ctx, &pb.GetPolicyValidationRequest{
+		ValidationId: "invalid-id",
+	})
+	g.Expect(err).To(HaveOccurred())
+}
+
+func TestListApplicationValidations(t *testing.T) {
+	g := NewGomegaWithT(t)
+
+	ctx := context.Background()
+
+	scheme, err := kube.CreateScheme()
+	g.Expect(err).To(BeNil())
+
+	ns1 := &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "weave-system",
+		},
+	}
+	ns2 := &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "default",
+		},
+	}
+
+	client := fake.NewClientBuilder().
+		WithScheme(scheme).
+		WithRuntimeObjects(ns1, ns2, makeValidationEvent(t, func(e *corev1.Event) {
+			e.ObjectMeta.Name = "Missing Owner Label - fake-event-2"
+			e.InvolvedObject.Namespace = "weave-system"
+			e.ObjectMeta.Namespace = "weave-system"
+			e.InvolvedObject.Name = "app1"
+			e.InvolvedObject.Kind = "HelmRelease"
+			e.Annotations["policy_name"] = "Missing Owner Label"
+			e.Annotations["policy_id"] = "weave.policies.missing-app-label"
+			e.Labels["pac.weave.works/id"] = "56701548-12c1-4f79-a09a-a12979904"
+		})).
+		WithIndex(&corev1.Event{}, "type", client.IndexerFunc(func(o client.Object) []string {
+			event := o.(*corev1.Event)
+			return []string{event.Type}
+		})).
+		Build()
+
+	cfg := makeServerConfig(client, t, "")
+	c := makeServer(cfg, t)
+	res, err := c.ListPolicyValidations(ctx, &pb.ListPolicyValidationsRequest{
+		Application: "app1",
+		Kind:        "HelmRelease",
+		Namespace:   "weave-system",
+		ClusterName: "Default",
+	})
+	g.Expect(err).NotTo(HaveOccurred())
+	g.Expect(err).To(BeNil())
```

Comment:
It would be better to add a check for res that it's not null.

Reply:
checking for null pointer will raise an error